x/feegrant/filtered_fee_test.go

```diff
@@ -0,0 +1,193 @@
+package feegrant_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
+	ocproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+func TestFilteredFeeValidAllow(t *testing.T) {
+	app := simapp.Setup(t, false)
+
+	ctx := app.BaseApp.NewContext(false, ocproto.Header{
+		Time: time.Now(),
+	})
+	eth := sdk.NewCoins(sdk.NewInt64Coin("eth", 10))
+	atom := sdk.NewCoins(sdk.NewInt64Coin("atom", 555))
+	smallAtom := sdk.NewCoins(sdk.NewInt64Coin("atom", 43))
+	bigAtom := sdk.NewCoins(sdk.NewInt64Coin("atom", 1000))
+	leftAtom := sdk.NewCoins(sdk.NewInt64Coin("atom", 512))
+	now := ctx.BlockTime()
+	oneHour := now.Add(1 * time.Hour)
+
+	// msg we will call in the all cases
+	call := banktypes.MsgSend{
+		FromAddress: "",
+		ToAddress: "",
+		Amount: sdk.Coins{},
+	}
+	cases := map[string]struct {
+		allowance *feegrant.BasicAllowance
+		msgs      []string
+		// all other checks are ignored if valid=false
+		fee       sdk.Coins
+		blockTime time.Time
+		valid     bool
+		accept    bool
+		remove    bool
+		remains   sdk.Coins
+	}{
+		"msg contained": {
+			allowance: &feegrant.BasicAllowance{},
+			msgs:      []string{sdk.MsgTypeURL(&call)},
```

Comment:
I think it could be more straight-forward to use the variable `call`, because we don't need to compare the msgs in the cases with the actual calling msg in the loop. To be more pedantic, I have to make sure that the msg type of "msg not contained" case is not identical to that of `call`, but I think it's not necessary.